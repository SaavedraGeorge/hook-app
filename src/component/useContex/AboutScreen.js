import React, { useContext } from 'react';
import { UserContext } from './UseContext';

export const AboutScreen = () => {
    const {setUser} = useContext(UserContext);
    const logOut = () =>{
        setUser({})
    }
    return (
        <div>
            <h1>About</h1>
            <hr />
            <button onClick={ logOut }> Logout</button>
        </div>
    );
};
