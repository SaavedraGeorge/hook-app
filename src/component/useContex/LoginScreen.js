import React, { useContext } from 'react';
import { UserContext } from './UseContext';

export const LoginScreen = () => {
    const {setUser} = useContext(UserContext);

    const cambio = () =>{
        setUser(
            {
                id: 456,
                name: 'alfa'
            }
        )
    }
    return (
        <div>
            <h1>Login</h1>
            <hr />
            <button className="btn btn-outline-primary" onClick={ cambio } >Cambio User</button>
        </div>
    );
};
