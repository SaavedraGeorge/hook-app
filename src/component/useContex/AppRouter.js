import React from 'react';
import {
    BrowserRouter as Router,
    Routes,
    Route
  } from "react-router-dom";
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import { NavBar } from './NavBar';

export const AppRouter = () => {
  return (
    <Router>
        <div>
            <NavBar />
            <Routes>
                <Route exact path="/" element={ <HomeScreen /> } />
                <Route exact path="/about" element={ <AboutScreen /> } />
                <Route exact path="/login" element={ <LoginScreen /> } />
                <Route path="*" element={
                        <div style={{ padding: "1rem" }}>
                        <p>Pagina no encontrada</p>
                        </div>
                } />
            </Routes>
        </div>
    </Router>
  );
};
