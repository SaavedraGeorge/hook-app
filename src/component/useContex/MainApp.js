import React, { useState } from 'react';
import { AppRouter } from './AppRouter';
import { UserContext } from './UseContext';

export const MainApp = () => {
    
    const [user, setUser] = useState({
        id: 1,
        name: 'jorge'
    });
    
    return (
            <UserContext.Provider value={{
                user,
                setUser
                
            }}>
                <AppRouter />
            </UserContext.Provider>
       
    );
};
