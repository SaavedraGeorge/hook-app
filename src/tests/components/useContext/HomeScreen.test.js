//configure se llama para llamar mount en reat 17
import {mount,  configure} from "enzyme";
//adaptaer es para llamar mount en reat 17
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { HomeScreen } from '../../../component/useContex/HomeScreen';
import { UserContext } from '../../../component/useContex/UseContext';

configure({adapter: new Adapter()});

describe('pruebas a <HomeScreen />', () => {
    const user = {
        name: 'jorge',
        email: 'asd@asd.com'
    }
    
    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }}>
            <HomeScreen />
        </UserContext.Provider>
        
    );
    
    test('should show componen corectly', () => {
        expect(wrapper).toMatchSnapshot();
    });
  
});
