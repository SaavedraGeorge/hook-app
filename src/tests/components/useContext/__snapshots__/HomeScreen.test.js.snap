// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas a <HomeScreen /> should show componen corectly 1`] = `
<div>
  <h1>
    Home
  </h1>
  <hr />
  <pre>
    {
 "name": "jorge",
 "email": "asd@asd.com"
}
  </pre>
</div>
`;
